### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by andriy-shymkiv**
[DEMO LINK](https://andriy-shymkiv.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -8,7 +8,56 @@ import { TodoFilter } from './components/TodoFilter';
...
+        return todo;
+
+      case 'Active':
+        return !todo.completed;
+
+      case 'Completed':
+        return todo.completed;
+
+      default:
+        return 0;
```
Comment (iryna-lelyk) [inline src/App.tsx:46]: Better to return just `todo` by default.

File: src/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -1,100 +1,96 @@
...
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+      <tbody>
+        {todos.map(todo => (
+          <tr
+            data-cy="todo"
+            className=""
```
Comment (iryna-lelyk) [inline src/components/TodoList/TodoList.tsx:38]: When todo, you're mapping is equal to selectedTodo, you should add class `has-background-info-light`

File: src/components/TodoModal/TodoModal.tsx [hunk 3]
```diff
@@ -1,12 +1,51 @@
...
+
+    getUserFromServer();
+  }, []);
+
+  const findClickedTodoBySelectedId = useCallback((todoId: number) => {
+    const findedTodo = todos.find(todo => todo.id === todoId);
+
+    return findedTodo;
+  }, []);
+
```
Comment (iryna-lelyk) [inline src/components/TodoModal/TodoModal.tsx:43]: You can save selected todo, not only id of it. And if you've saved whole todo, you would have access to userId also

**Review by iryna-lelyk — APPROVED** [review]
Good job!

File: src/App.tsx [hunk 4]
```diff
@@ -8,7 +8,56 @@ import { TodoFilter } from './components/TodoFilter';
...
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+
+import { getTodos } from './api';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [userId, setUserId] = useState<number>(0);
+  const [query, setQuery] = useState<string>('');
```
Comment (mykola-koval) [inline src/App.tsx:18]: don't need the generic here, actually
it is automatically calculated based on the default value
```suggestion
  const [query, setQuery] = useState('');
```

File: src/components/TodoList/TodoList.tsx (hunk 2 above)
Comment (mykola-koval) [reply src/components/TodoList/TodoList.tsx:38]: agree, add it

**Pull request closed without merging**